### PR TITLE
Fixes various bugs picked up by an OpenDream compile attempt

### PR DESCRIPTION
**Pull request by Altoids1**
## Summary
Hi guys, I'm back for a second.

I tried to compile Yogstation on OD again. I've been working on improving the const-folding available, and this resulted in OD picking up a lot of broken arithmetic and switch statements scattered about the place, which under BYOND just mysteriously runtimes or causes strange behaviour silently.

So, guess what, here's a bunch of fixes for that!

## Changelog

:cl:  Altoids
bugfix: Fixed non-bloodsuckers not consistently showing that they have pale skin.
bugfix: Fixed some sorts of bloodsucking not reducing blood volume.
bugfix: Fixed some wines not always displaying their taste correctly.
bugfix: Fixed the King Goat plushie maybe not correctly eating his cabbage.
/:cl:

### CODE REVIEW

File: yogstation/code/game/objects/items/plushes.dm [hunk 1]
```diff
@@ -138,7 +138,6 @@
 		new/obj/item/toy/plush/goatplushie/angry/guardgoat(location)
 
 /obj/item/toy/plush/goatplushie/angry/kinggoat/attackby(/obj/item/reagent_containers/food/snacks/grown/cabbage/C,mob/living/user,params)
```
Comment (ToasterBiome) [inline yogstation/code/game/objects/items/plushes.dm]: cant you shove malformed params in here? Sus!

Comment (Altoids1) [reply yogstation/code/game/objects/items/plushes.dm]: Oh wait

so with that slash there, I think this becomes a global var: https://www.byond.com/forum/post/2830750

**Review by ReddicusDragon — APPROVED** [review]
it compiles, good enough for me.

**Review by JamieD1 — APPROVED** [review]
Cheers

**Pull request merged**